sip/execution_control/configuration_db/redis/db_client/processing_controller_client.py

```diff
@@ -1,389 +1,257 @@
-                                              current date.
-            project (str, optional): Project
+            block_id (str): Scheduling block instance or processing block id
 
         Returns:
-            str, A valid Scheduling Block Instance ID.
+            str, db key for the specified scheduling block instance or
+            processing block event data.
 
         """
-        if date is None:
-            date = datetime.datetime.utcnow()
+        return aggregate_events_list(self.get_key(block_id))
 
-        if isinstance(date, str):
-            _date = date
-        elif isinstance(date, datetime.datetime):
-            _date = date.strftime('%Y%m%d')
-        else:
-            raise RuntimeError('Unknown date type.')
+    def get_event_data_key(self, block_id: str) -> str:
+        """Get event data db key.
 
-        sbi_index = self.get_num_scheduling_blocks()
-        return '{}-{}-sbi{:03d}'.format(_date, project, sbi_index)
+        Return the scheduling block instance or processing block events
+        data db key.
 
-    def get_scheduling_block_ids(self):
-        """Get the sorted list of Scheduling Block Instance ids.
+        Args:
+            block_id (str): Scheduling block instance or processing block id
 
         Returns:
-            list, All Scheduling Block Instance IDs in the database
+            str, db key for the specified scheduling block instance or
+            processing block event data.
 
         """
-        scheduling_block_ids = []
+        return aggregate_events_data(self.get_key(block_id))
 
-        # Pattern used to search scheduling block ids
-        pattern = 'scheduling_block:*'
-        block_ids = self._db.get_keys(pattern)
+    def get_events(self, block_id: str) -> List[events.Event]:
+        """Get event data.
 
-        for block_id in block_ids:
-            if 'processing_block' not in block_id:
-                id_split = block_id.split(':')[-1]
-                scheduling_block_ids.append(id_split)
-        return sorted(scheduling_block_ids)
+        Get event data for the specified Scheduling block instance or
+        processing block.
 
-    def get_num_scheduling_blocks(self):
-        """Get the number of Scheduling Blocks Instance ids in the database.
+        Args:
+            block_id (str): Scheduling block instance or processing block id
 
         Returns:
-            int, The number of Scheduling Block Instance IDs
+            list of event data dictionaries
 
         """
-        return len(self.get_scheduling_block_ids())
+        event_data_key = self.get_event_data_key(block_id)
+        event_list = []
+        for event_id, data in self._db.get_hash_dict(event_data_key).items():
+            data = ast.literal_eval(data)
+            event_list.append(events.Event(event_id, self._aggregate_type, '',
+                                           data))
+        return event_list
+
+    def get_key(self, block_id: str,) -> str:
```

Comment:
Looks like there is a stray comma after the last argument and just before the closing bracket?